src/base/memory_monitor_sr.cpp

```diff
@@ -76,6 +86,128 @@ namespace cubmem
+	    assert (false);
+	  }
+	else
+	  {
+	    sprintf (debug_search->second.filename, "%s", file);
+	    debug_search->second.line = line;
+	    debug_search->second.is_exist = true;
+	  }
+      }
+    else
+      {
+	MMON_DEBUG_INFO debug_info;
+
+	sprintf (debug_info.filename, "%s", file);
+	debug_info.line = line;
+	debug_info.is_exist = true;
+
+	std::pair<tbb::concurrent_unordered_map<intptr_t, MMON_DEBUG_INFO>::iterator, bool> debug_insert;
+	debug_insert = m_error_tracking_map.insert (std::pair <intptr_t, MMON_DEBUG_INFO> (ptr_key, debug_info));
+	if (!debug_insert.second)
+	  {
+	    // Case2. Double allocation (unreachable)
+           //    This case is not reached in normal memory allocation situations and
+           //    indicates a problem in the default allocation mechanism,
+           //    such as malloc(), if it is reached.
+	    fprintf (stderr, "double memory allocation is occurred\n");
+	    fflush (stderr);
+	    assert (false);
+	  }
+      }
+  }
+
+  void memory_monitor::check_sub_stat_tracking_error_is_exist (MMON_METAINFO *metainfo)
+  {
+    intptr_t ptr_key = reinterpret_cast <intptr_t> (metainfo);
+    auto debug_search = m_error_tracking_map.find (ptr_key);
+    if (debug_search != m_error_tracking_map.end())
+      {
+	if (debug_search->second.is_exist)
+	  {
+	    debug_search->second.is_exist = false;
+	  }
+	else
+	  {
+	    if (metainfo->magic_number == m_magic_number)
+	      {
+		// Case3. Metainfo corrupted
+               //    This case indicates that the metadata information owned
+               //    by the memory_monitor class for tracking has been corrupted.
+               //    This can occur in two scenarios:
+               //
+               //      - Overflow occurring in the memory allocated through cub_alloc(),
+               //      corrupting the metadata space.
+               //      - Memory allocated through cub_alloc() is deallocated without
+               //      erasing the metadata information via default free(), and then
+               //      deallocated through cub_free after being reallocated through basic
+               //      allocation functions like malloc().
+               //
+               //    In the second scenario, even if a different size is allocated compared
+               //    to the first allocation, the position of the pointer storing the metadata
+               //    information may not change. The position of the metadata is determined by
+               //    malloc_usable_size(), which can return a larger value than the size
+               //    requested by the user. This is because the OS allocates memory in chunks
+               //    rather than exactly as much as the user requested. Thus, even if the user
+               //    receives the same chunk through the basic allocation function,
+               //    the amount of memory usable by the user can be larger, indicating
+               //    potential corruption of the metadata space.
```

Comment:
```suggestion
		//    This case indicates that the metadata information owned
		//    by the memory_monitor class for tracking has been corrupted.
		//    This can occur in two scenarios:
		//
		//      - Overflow occurring in the memory allocated through cub_alloc(),
		//      corrupting the metadata space.
		//      - Memory allocated through cub_alloc() is deallocated without
		//      erasing the metadata information via default free(), and then
		//      deallocated through cub_free after being reallocated through basic
		//      allocation functions like malloc().
		//
		//    In the second scenario, even if a different size is allocated compared
		//    to the first allocation, the position of the pointer storing the metadata
		//    information may not change. The position of the metadata is determined by
		//    malloc_usable_size(), which can return a larger value than the size
		//    requested by the user. This is because the OS allocates memory in chunks
		//    rather than exactly as much as the user requested. Thus, even if the user
		//    receives the same chunk through the basic allocation function,
		//    the amount of memory usable by the user can be larger, indicating
		//    potential corruption of the metadata space.
```